src/cordova/plugin/OutlineAndroidLib/outline/build.gradle

```diff
@@ -0,0 +1,49 @@
+plugins {
+    id 'com.android.library'
+}
+
+android {
+    namespace 'org.outline'
+    compileSdk 30
+
+    defaultConfig {
+        minSdk 22
+        targetSdk 30
+
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+        consumerProguardFiles "consumer-rules.pro"
+    }
+
+    buildTypes {
+        release {
+            minifyEnabled true
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+        }
+        debug {
+        }
+    }
+    compileOptions {
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+}
+
+dependencies {
+
+    // To make this project editable directly in the source tree, we need to load
+    // tun2socks.aar from its original path in the third_party directory.  We do
+    // this by relative path:
+    compileOnly fileTree(dir: '../../../../../third_party/outline-go-tun2socks/android', include: ['tun2socks.aar'])
+    // However, Cordova uses this library as a subproject of plugins/cordova-plugin-outline/
+    // which results in it being built from a different path, so we need a different
+    // relative path to locate tun2socks.aar:
+    compileOnly fileTree(dir: '../../../../third_party/outline-go-tun2socks/android', include: ['tun2socks.aar'])
```

Comment:
It seems like we can use a `<framework> with `custom="true" type="gradleReference"` in the plugin.xml. I wonder if that helps.

Reply:
There's a broken abstraction here. This library should standalone, but is being affected by how the plugin works. We shouldn't need these two entries.

Can you remove the entry for Cordova? Also move the lib outside the plugin, into src/cordova/android/OutlineAndroidLib.

In the plugin, specify a repository that points to src/cordova/android. See comments in the plugin code. OutlineAndroidLib and Tun2socks can be treated just like external dependencies that way.

Reply:
I don't know how to do this using a repository that point to `src/cordova/android`, but I was able to accomplish the same result using a [composite build](https://docs.gradle.org/current/userguide/composite_builds.html).